lib/clients/client.js

```diff
@@ -102,6 +98,22 @@ BaseAPIClient.prototype.registerDataStore = function registerDataStore(dataStore
 
 
 /**
+ * Empties the request queue and returns it to a clean slate.
+ */
+BaseAPIClient.prototype.clearQueue = function clearQueue() {
+  this.logger('info', 'Clearing the request queue');
+  this.requestQueue.kill();
+
+  // Initially tried to accomplish this with just `requestQueue.kill`, but it leaves the queue
+  // blocked. See https://github.com/caolan/async/issues/1422.
+  this.requestQueue = async.queue(
```

Comment:
there's some code duplication here with L61-L63. but honestly, that can be cleaned up with a later pass.